include/taco/lower/mode_format_impl.h

```diff
@@ -172,6 +172,8 @@ class ModeFormatImpl {
   const bool hasInsert;
   const bool hasAppend;
 
+  bool hasFixedSize;
```

Comment:
Can perhaps replace this and next member with a `Dimension` object? This should also be made const.